projects/plugins/jetpack/functions.opengraph.php

```diff
@@ -99,7 +99,7 @@ function jetpack_og_tags() {
 		$tags['og:title'] = wp_get_document_title();
 
 		$archive = get_queried_object();
-		if ( ! empty( $archive ) ) {
+		if ( $archive instanceof WP_Term ) {
```

Comment:
Given the `is_post_type_archive()` below, maybe `$archive` can be `WP_Post_Type` too?

Reply:
Ah yes, I see your point. For some reason, 'WP_Post_Type' object passes the condition `if ( is_category() || is_tag() || is_tax() )` and causes the error I was looking to solve. Maybe we should be editing the logic to something like:
```
		if ( $archive instanceof WP_Term ) {
			if ( is_category() || is_tag() || is_tax() ) {
				$tags['og:url']         = get_term_link( $archive->term_id, $archive->taxonomy );
				$tags['og:description'] = $archive->description;
			}
		} elseif ( is_post_type_archive() ) {
				$tags['og:url']         = get_post_type_archive_link( $archive->name );
				$tags['og:description'] = $archive->description;
		}
```

I'm not sure if we even need the `if ( is_category() || is_tag() || is_tax() )` condition in that case. What do you think?

Reply:
> I'm not sure if we even need the if ( is_category() || is_tag() || is_tax() ) condition in that case. What do you think?

Yeah, you should be able to simplify it that way (although it may still be necessary to check if `$archive` is empty before you check if `$archive instanceof WP_Term`).

Reply:
I updated the code in [`f7b2cdb` (#39749)](https://github.com/Automattic/jetpack/pull/39749/commits/f7b2cdbd89cb16e1701c1f315a774470b1931bb8)

>although it may still be necessary to check if $archive is empty before you check if $archive instanceof WP_Term

@anomiex said above: "instanceof will already do the right thing if it's null, false, or the like."